### PR TITLE
angie: init at 1.4.0

### PR DESCRIPTION
**Pull request by Izorkin**
## Description of changes
Add angie package version 1.4.0.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
Update package to version 1.3.2

**Review by SuperSandro2000 — APPROVED** [review]

File: nixos/modules/services/web-servers/nginx/default.nix [hunk 1]
```diff
@@ -1144,18 +1144,20 @@ in
       }
 
       {
-        assertion = cfg.package.pname != "nginxQuic" -> !(cfg.enableQuicBPF);
+        assertion = cfg.package.pname != "nginxQuic" && cfg.package.pname != "AngieQuic" -> !(cfg.enableQuicBPF);
```
[the remark this answers is not in the document]
Comment (Izorkin) [reply nixos/modules/services/web-servers/nginx/default.nix:1147]: I hope that after official support for QUIC in OpenSSL, the number of variants will decrease.

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply nixos/modules/services/web-servers/nginx/default.nix:1147]: me, too.

**Comment by Izorkin** [comment]
Resolved conflicts.

**Comment by Izorkin** [comment]
@GrahamcOfBorg eval

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 268806` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>angie</li>
    <li>angie-console-light</li>
    <li>angie-console-light.doc</li>
    <li>angie.doc</li>
    <li>angieQuic</li>
    <li>angieQuic.doc</li>
    <li>devpi-client</li>
    <li>devpi-client.dist</li>
    <li>devpi-server</li>
    <li>devpi-server.dist</li>
    <li>nginx (nginxStable)</li>
    <li>nginx.doc (nginxStable.doc)</li>
    <li>nginxMainline</li>
    <li>nginxMainline.doc</li>
    <li>nginxQuic</li>
    <li>nginxQuic.doc</li>
    <li>nginxShibboleth</li>
    <li>nginxShibboleth.doc</li>
    <li>openresty</li>
    <li>openresty.doc</li>
  </ul>
</details>

All tests pass and diff LGTM.

Would it make sense to advertise `angie` and the Quic variants in the description of `services.nginx.package`?

The proposed changes where i change the test parametrization from strings to derivations are not _required_, just a suggestion. If you do chose to adopt them, please also do so in `nixos/tests/nginx-variants.nix`.

**Comment by Izorkin** [comment]
> Would it make sense to advertise `angie` and the Quic variants in the description of `services.nginx.package`?

This variant?
```diff
diff --git a/nixos/modules/services/web-servers/nginx/default.nix b/nixos/modules/services/web-servers/nginx/default.nix
index e4f02692daed..2d61b6d0f909 100644
--- a/nixos/modules/services/web-servers/nginx/default.nix
+++ b/nixos/modules/services/web-servers/nginx/default.nix
@@ -649,6 +649,9 @@ in
           Nginx package to use. This defaults to the stable version. Note
           that the nginx team recommends to use the mainline version which
           available in nixpkgs as `nginxMainline`.
+          To support the HTTP/3 protocol, must use the `nginxQuic` package.
+          Can also use Nginx forks - `angie`, `openresty` and `tengine`
+          packages.
         '';
       };

```

> The proposed changes where i change the test parametrization from strings to derivations are not _required_, just a suggestion. If you do chose to adopt them, please also do so in `nixos/tests/nginx-variants.nix`.

Update PR.

**Comment by pbsds** [comment]
Reworded somewhat

```diff
diff --git a/nixos/modules/services/web-servers/nginx/default.nix b/nixos/modules/services/web-servers/nginx/default.nix
index e4f02692daed..2d61b6d0f909 100644
--- a/nixos/modules/services/web-servers/nginx/default.nix
+++ b/nixos/modules/services/web-servers/nginx/default.nix
@@ -649,6 +649,9 @@ in
           Nginx package to use. This defaults to the stable version. Note
           that the nginx team recommends to use the mainline version which
           available in nixpkgs as `nginxMainline`.
+          Supported Nginx forks include `angie`, `openresty` and `tengine`.
+          For HTTP/3 support, use `nginxQuic` or `angieQuic`.
         '';
       };
```

**Comment by Izorkin** [comment]
@pbsds thanks, update.

**Review by pbsds — APPROVED** [review]
waiting for ofborg

**Comment by pbsds** [comment]
```
$ nom-build nixpkgs/ -A angie.passthru.tests -A angie-console-light.passthru.tests -A nginx.passthru.tests
error: attribute 'angie' missing

       at /dev/shm/nixpkgs-review/pr-268806-1/nixpkgs/pkgs/servers/http/angie/default.nix:35:13:

           34|   passthru.tests = {
           35|     angie = nixosTests.nginx-variants.angie;
             |             ^
           36|     angie-api = nixosTests.angie-api;
       Did you mean Angie?
```

**Comment by Izorkin** [comment]
Fixed:
```diff
diff --git a/pkgs/servers/http/angie/default.nix b/pkgs/servers/http/angie/default.nix
index 1bf6ac7e031a..05fc6ba670fa 100644
--- a/pkgs/servers/http/angie/default.nix
+++ b/pkgs/servers/http/angie/default.nix
@@ -10,7 +10,7 @@

 callPackage ../nginx/generic.nix args rec {
   version = "1.4.0";
-  pname = if withQuic then "AngieQuic" else "Angie";
+  pname = if withQuic then "angieQuic" else "angie";

   src = fetchurl {
     url = "https://download.angie.software/files/angie-${version}.tar.gz";
```

**Comment by pbsds** [comment]
@ofborg test nginx-etag

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 268806` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>angie</li>
    <li>angie-console-light</li>
    <li>angie-console-light.doc</li>
    <li>angie.doc</li>
    <li>angieQuic</li>
    <li>angieQuic.doc</li>
    <li>devpi-client</li>
    <li>devpi-client.dist</li>
    <li>devpi-server</li>
    <li>devpi-server.dist</li>
    <li>nginx</li>
    <li>nginx.doc</li>
    <li>nginxMainline</li>
    <li>nginxMainline.doc</li>
    <li>nginxQuic</li>
    <li>nginxQuic.doc</li>
    <li>nginxShibboleth</li>
    <li>nginxShibboleth.doc</li>
    <li>openresty</li>
    <li>openresty.doc</li>
  </ul>

I also ran the tests:

```
/nix/store/6hgr341lax655n183r7m64mp5x5l9am9-vm-test-run-acme
/nix/store/fminlc0nr82gyi4wqbjxw9ib136aqs0c-vm-test-run-nginx-variant-angie
/nix/store/64p0pnmdcz9qczgnw2ghjvr62b5jc9yw-vm-test-run-angie-api
/nix/store/75a0y1kzykm6v0hkdp19qgmpqmafsvyv-vm-test-run-nginx-http3-angieQuic
/nix/store/8vcxximzd6xnmfd17024v90d4vk6pza7-vm-test-run-nginx
/nix/store/v73r3cddy25sax1n0gpnll3n507zr1rk-vm-test-run-nginx-auth
/nix/store/mnm22rs3lsmrhcaghl306l9k91g0f41a-vm-test-run-nginx-etag
/nix/store/zh1dbfcf358qr5pfxyzi6kp4cjwjiw30-vm-test-run-nginx-globalredirect
/nix/store/4vmcaf53dqhz5j36m4sivcr07hvin32p-vm-test-run-nginx-proxyprotocol
/nix/store/6xiz51mbl84d7wjzw7ds9s5qvzp63rsq-vm-test-run-nginx-pubhtml
/nix/store/95dbbcm5vqlzbxdibld6pvn7z3hakw5m-vm-test-run-nginx-sso
/nix/store/wrdl82vnggzdlzbmnlfxmwyshwg3rv48-vm-test-run-nginx-status-page
/nix/store/lsd4pxkkp72bn45k91hsbhh11mij0np2-vm-test-run-nginx-unix-socket
/nix/store/kd6dhnxay4w665srbwwmj75gqiqs7wg8-vm-test-run-nginx-variant-angieQuic
/nix/store/8lkisx1rmws77dhxykdi13c2phssa09d-vm-test-run-nginx-variant-nginx
/nix/store/p0r4pfc6w0ywyjxrwqvj4mw8lqwhxjk6-vm-test-run-nginx-variant-nginxQuic
/nix/store/hc9597mwfmnz3zyvd8vj6inmx9na442i-vm-test-run-nginx-variant-openresty
/nix/store/p649fvjz1i1lg0gsbn75cg7r1jgzbyhx-vm-test-run-nginx-variant-tengine
```

</details>


LGTM, waiting for ofborg

**Comment by Izorkin** [comment]
@pbsds as i noticed, the ёnginx-etagё test is unstable, sometimes you have to restart it.

**Comment by Izorkin** [comment]
Do we have to change anything else to merge pr?

**Comment by pbsds** [comment]
Thanks!

**Pull request merged**